frontend/src/metabase-types/store/entities.ts

```diff
@@ -19,7 +19,7 @@ import {
 
 export interface EntitiesState {
   actions?: Record<WritebackActionId, WritebackAction>;
-  collections?: Record<CollectionId, Collection>;
+  collections?: Partial<Record<CollectionId, Collection>>;
```

Comment:
What does `Partial` solve here?

Reply:
I believe you want to use `Record<CollectionId, Partial<Collection>>` instead?

This will ensure the state always looks something like this:
```js
{
  1: collection,
  2: collectionWithoutRootOrPersonal
}
```

But `Partial<Record<CollectionId, Collection>>` could give you:
```js
{
  1: undefined
  2: collection
}
```

unless I misread this.

Reply:
Gotcha, I think it might make sense to leave a comment about it